CONTRIBUTING.md

```diff
@@ -55,55 +60,144 @@ Prerequisites:
 Steps:
 
 - Clone repository
-- Open it in editor
+- Open it in your editor
 - For VS Code, execute `Dev Containers: Reopen in container` command
 
 For detailed instructions, please see the [Dev Containers tutorial](https://code.visualstudio.com/docs/devcontainers/tutorial).
 
-### Extra tidbits
+### 🧱 Structure of the Repository
+
+If you're a newcomer to Gradio, we recommend getting familiar with the overall structure of the repository so that you can focus on the part of the source code you'd like to contribute to.
 
-- You can run gradio scripts in reload mode which will watch for changes in the `gradio` folder and reload the app if changes are made.
+- `/gradio`: contains the Python source code for the library
+  - `/gradio/interface.py`: contains the Python source code for the core `Interface` class
+  - `/gradio/blocks.py`: contains the Python source code for the core `Blocks` class
+  - `/gradio/components.py`: contains the Python source code for the `components`, you can add your custom components here.
+- `/js`: contains the HTML/JS/CSS source code for the library ([start here for frontend changes](/js/README.md))
+- `/test`: contains Python unit tests for the library
+- `/demo`: contains demos that are used in the documentation, you can find `Gradio` examples over here.
+- `/website`: contains the code for the Gradio website (www.gradio.app). See the README in the `/website` folder for more details
+
+
+
+- To develop the frontend app, you should also follow [js/README.md](js/README.md).
+
+### 🚀 Run a Gradio app
+
+You can get started by creating an `app.py` file in the root:
+
+```
+import gradio as gr
+
+with gr.Blocks() as demo:
+   gr.Button()
+    
+if __name__ == "__main__":
+   demo.launch()
+```
+
+then run:
 
 ```
 gradio app.py
 ```
 
-- To develop the frontend app, you should also follow [js/README.md](js/README.md).
+This will start the backend server in reload mode, which will watch for changes in the `gradio` folder and reload the app if changes are made.
+
+If you're making frontend changes, start the frontend server:
+
+```
+pnpm dev
+```
+Any changes to the frontend will also reload automatically in the browser. 
```

Comment:
Should we add that this launches a separate browser tab running on port 9876 by default, which loads the backend from 7860?